src/sc2laddercore/LadderGame.cpp

```diff
@@ -6,359 +22,36 @@
-            }
-        }
-        return CurrentExitCase;
-    }
-    catch (const std::exception& e)
-    {
-        PrintThread{} << e.what() << std::endl;
-        return ExitCase::ClientTimeout;
-    }
-}
-
-void ResolveMap(const std::string& map_name, SC2APIProtocol::RequestCreateGame* request, sc2::ProcessSettings process_settings) {
-    // BattleNet map
-    if (!sc2::HasExtension(map_name, ".SC2Map")) {
-        request->set_battlenet_map_name(map_name);
-        return;
-    }
-
-    // Absolute path
-    SC2APIProtocol::LocalMap* local_map = request->mutable_local_map();
-    if (sc2::DoesFileExist(map_name)) {
-        local_map->set_map_path(map_name);
-        return;
-    }
-
-    // Relative path - Game maps directory
-    std::string game_relative = sc2::GetGameMapsDirectory(process_settings.process_path) + map_name;
-    if (sc2::DoesFileExist(game_relative)) {
-        local_map->set_map_path(map_name);
-        return;
-    }
-
-    // Relative path - Library maps directory
-    std::string library_relative = sc2::GetLibraryMapsDirectory() + map_name;
-    if (sc2::DoesFileExist(library_relative)) {
-        local_map->set_map_path(library_relative);
-        return;
-    }
-
-    // Relative path - Remotely saved maps directory
-    local_map->set_map_path(map_name);
-}
-
-sc2::GameRequestPtr CreateStartGameRequest(const std::string &MapName, std::vector<sc2::PlayerSetup> players, sc2::ProcessSettings process_settings)
+LadderGame::LadderGame(int InCoordinatorArgc, char** InCoordinatorArgv, LadderConfig *InConfig)
+    : CoordinatorArgc(InCoordinatorArgc)
+    , CoordinatorArgv(InCoordinatorArgv)
+    , Config(InConfig)
 {
-    sc2::ProtoInterface proto;
-    sc2::GameRequestPtr request = proto.MakeRequest();
-
-    SC2APIProtocol::RequestCreateGame* request_create_game = request->mutable_create_game();
-    for (const sc2::PlayerSetup& setup : players)
-    {
-        SC2APIProtocol::PlayerSetup* playerSetup = request_create_game->add_player_setup();
-        playerSetup->set_type(SC2APIProtocol::PlayerType(setup.type));
-        playerSetup->set_race(SC2APIProtocol::Race(int(setup.race) + 1));
-        playerSetup->set_difficulty(SC2APIProtocol::Difficulty(setup.difficulty));
-    }
-    ResolveMap(MapName, request_create_game, process_settings);
-
-    request_create_game->set_realtime(false);
-    return request;
+    const int maxGameTimeInt = Config->GetIntValue("MaxGameTime");
+    MaxGameTime = maxGameTimeInt > 0 ? static_cast<uint32_t>(maxGameTimeInt) : 0;
+    const int MaxRealGameTimeInt = Config->GetIntValue("MaxRealGameTime");
+    MaxGameTime = MaxRealGameTimeInt > 0 ? static_cast<uint32_t>(MaxRealGameTimeInt) : 0;
+    RealTime = Config->GetBoolValue("RealTimeMode");
 }
 
-
 void LadderGame::LogStartGame(const BotConfig &Bot1, const BotConfig &Bot2)
 {
     std::time_t t = std::time(nullptr);
     std::tm tm = *std::localtime(&t);
+    // Why do we log this in stderr ?
```

Comment:
This is logged, as I wanted a way to differentiate whch game exactly caused an error, I might change this later, so it only logs if there is actually stdout error